RELEASE-NOTES.md

```diff
@@ -28,10 +28,23 @@ The Axom project release numbers follow [Semantic Versioning](http://semver.org/
 ###  Removed
 
 ###  Fixed
-- Added a guard for sidre-related mint API usage in a quest example
+
+## [Version 0.10.1] - Release date 2024-10-dd
```

Comment:
Please update the links at the bottom of this file:
```
[Unreleased]: https://github.com/LLNL/axom/compare/v0.10.1...develop
[Version 0.10.1]: https://github.com/LLNL/axom/compare/v0.10.0...v0.10.1
```

Reply:
D'oh! Thanks for catching that. Change pushed.